### ISSUE TITLE
K.in_train_phase()/K.in_test_phase() doesn't work in the custom loss function if the model has no _uses_learning_phase outputs

### ISSUE THREAD

**Issue by kitsnet** [keras-team/keras#8141]
Well, the topic is self-explanatory.

A short workaround for a functional-style model for use with such an objective function / loss function:

```
b._uses_learning_phase = True          # add this line
model = Model(inputs=a, outputs=b)
```

You can close the ticket if you don't want to fix it; I am just leaving it for people that may need this workaround.

**Comment by davideboschetto** [comment]
One quick note: setting _uses_learning_phase = True in a TimeDistributed layer with a model wrapped inside is needed to make sure that the wrapped model gets trained as well. Looks to be a bug, I guess.
